framework/contrib/ice_updater/include/Updater.h

```diff
@@ -333,7 +394,15 @@ class UpdaterThread: public tbb::task,
    * posts queue to url with cURL in JSON format.
    * The thread is then put to sleep for 1000 milliseconds.
    */
-  tbb::task* execute();
+  void execute() {
+	  processThread = MooseSharedPointer<std::thread>(new std::thread(&UpdaterThread::runThreaded, this));
```

Comment:
You don't necessarily have to use MooseSharedPointer if you require c++11. It's fine either way or you could use std::shared_pointer